Zero-shot AutoML (#468)
microsoft/FLAML: website/docs/Use-Cases/Zero-Shot-AutoML.md

```python
estimator = flaml.default.LGBMRegressor(default_location="location_for_defaults")
```

Or,

```python
preprocess_and_suggest_hyperparams(
    "classification", X_train, y_train, "lgbm", location="location_for_defaults"
)
```

Or,

```python
X_train, y_train = load_iris(return_X_y=True, as_frame=as_frame)
automl = AutoML()
automl_settings = {
    "task": "classification",
    "log_file_name": "test/iris.log",
    "starting_points": "data:location_for_defaults",
    "estimator_list": ["lgbm", "xgb_limitdepth", "rf"]
    "max_iter": 0,
}
automl.fit(X_train, y_train, **automl_settings)
```

Since this is a multiclass task, it will look for the following files under `{location_for_defaults}/`:

- `all/multiclass.json`.
- `{learner_name}/multiclass.json` for every learner_name in the estimator_list.

Read the next section to understand how to generate these files if you would like to meta-learn the defaults yourself.

## How to Prepare Offline

This section is intended for:
1. AutoML providers for a particular domain.
1. Data scientists or engineers who need to repeatedly train models for similar tasks with varying training data.

> Instead of running full hyperparmaeter tuning from scratch every time, one can leverage the tuning experiences in similar tasks before. While we have offered the meta-learned defaults from tuning experiences of several popular learners on benchmark datasets for classification and regression, you can customize the defaults for your own tasks/learners/metrics based on your own tuning experiences.

Reviewer: ```suggestion
Instead of running full hyperparameter tuning from scratch every time, one can leverage the tuning experiences in similar tasks before. While we have offered the meta-learned defaults from tuning experiences of several popular learners on benchmark datasets for classification and regression, you can customize the defaults for your own tasks/learners/metrics based on your own tuning experiences.
```

> ### Prepare a collectioin of training tasks

Reviewer: ```suggestion
### Prepare a collection of training tasks
```

```

In a few seconds, it will produce the following files as output:

- test/default/lgbm/binary.json: the learned defaults for lgbm.
- test/default/xgb_limitdepth/binary.json: the learned defaults for xgb_limitdepth.
- test/default/rf/binary.json: the learned defaults for rf.
- test/default/all/binary.json: the learned defaults for lgbm, xgb_limitdepth and rf together.

Change "binary" into "multiclass" or "regression", or your own types in your "results.csv" for the other types of tasks. To update the learned defaults when more experiences are available, simply update your input files and rerun the learning command.

### "Flamlize" a learner

You have now effectively built your own zero-shot AutoML solution. Congratulations!

Optionally, you can "flamlize" a learner using [`flaml.default.flamlize_estimator`](../reference/default/estimator#flamlize_estimator) for easy dissemination. For example,

```python
import sklearn.ensemble as ensemble
from flaml.default import flamlize_estimator
ExtraTreesClassifier = flamlize_estimator(
    ensemble.ExtraTreesClassifier, "extra_tree", "classification"
)
```

> Then, you can share this "flamlized" `ExtraTreesClassifier` together with the location of your learned defaults with others (or the _future_ yourself). They will benefit from your past experience. Your group can also share experiences in a central place and update the learned defaults continuously. Overtime, your organization gets better collectively.

Reviewer: ```suggestion
Then, you can share this "flamlized" `ExtraTreesClassifier` together with the location of your learned defaults with others (or the _future_ yourself). They will benefit from your past experience. Your group can also share experiences in a central place and update the learned defaults continuously. Over time, your organization gets better collectively.
```
